src/foundry/client/applications/ux/drag-drop.d.mts

```diff
@@ -12,10 +14,141 @@
  * });
  * dragDrop.bind(html);
  * ```
- * @remarks TODO: Stub, copy from v12 implementation & update
  */
-declare class DragDrop {}
+declare class DragDrop {
+  /**
+   * @param options - (default: `{}`)
+   */
+  constructor({ dragSelector, dropSelector, permissions, callbacks }?: DragDrop.Configuration);
+
+  /**
+   * A set of callback functions for each action of the drag & drop workflow.
+   * @defaultValue `{}`
+   */
+  callbacks: InexactPartial<Record<DragDrop.Action, (event: DragEvent) => void>>;
+
+  /**
+   * The HTML selector which identifies draggable elements.
+   * @defaultValue `undefined`
+   */
+  dragSelector: string | null | undefined;
+
+  /**
+   * The HTML selector which identifies drop targets.
+   * @defaultValue `undefined`
+   */
+  dropSelector: string | null | undefined;
+
+  /**
+   * A set of functions to control authorization to begin drag workflows, and drop content.
+   * @defaultValue `{}`
+   */
+  permissions: Partial<Record<DragDrop.PermissionKey, (selector: this["dragSelector"]) => boolean>>;
+
+  /**
+   * Bind the DragDrop controller to an HTML application
+   * @param html - The HTML element to which the handler is bound
+   */
+  bind(html: HTMLElement): this;
+
+  /**
+   * Execute a callback function associated with a certain action in the workflow
+   * @param event  - The drag event being handled
+   * @param action - The action being attempted
+   */
+  callback(event: DragEvent, action: DragDrop.Action): void;
+
+  /**
+   * Test whether the current user has permission to perform a step of the workflow
+   * @param action   - The action being attempted
+   * @param selector - The selector being targeted
+   * @returns Can the action be performed?
+   */
+  can(action: DragDrop.PermissionKey, selector?: this["dragSelector"]): boolean;
+
+  /**
+   * Handle the start of a drag workflow
+   * @param event - The drag event being handled
+   * @internal
+   */
+  protected _handleDragStart(event: DragEvent): void;
+
+  /**
+   * Handle a drag workflow ending for any reason.
+   * @param event - The drag event.
+   */
+  protected _handleDragEnd(event: DragEvent): void;
+
+  /**
+   * Handle entering a drop target while dragging.
+   * @param event - The Drag event.
+   */
+  protected _handleDragEnter(event: DragEvent): void;
+
+  /**
+   * Handle leaving a drop target while dragging.
+   * @param event - The drag event.
+   */
+  protected _handleDragLeave(event: DragEvent): void;
+
+  /**
+   * Handle a dragged element over a droppable target
+   * @param event - The drag event being handled
+   * @internal
+   */
+  protected _handleDragOver(event: DragEvent): false;
```

Comment:
Foundry doesn't document a return, therefore I'd make this `void` or at best `void | false`